compiler_opt/rl/log_reader.py

```diff
@@ -165,38 +143,34 @@ def __getitem__(self, index: int):
 
 
 @dataclasses.dataclass(frozen=True)
-class Header:
-  features: List[TensorSpec]
-  score: Optional[TensorSpec]
+class _Header:
+  features: List[tf.TensorSpec]
+  score: Optional[tf.TensorSpec]
 
 
-def read_tensor(fs: BinaryIO, ts: TensorSpec) -> TensorValue:
-  size = math.prod(ts.shape) * ctypes.sizeof(ts.element_type)
+def _read_tensor(fs: BinaryIO, ts: tf.TensorSpec) -> _TensorValue:
+  size = math.prod(ts.shape) * ctypes.sizeof(_dtype_to_ctype[ts.dtype])
   data = fs.read(size)
-  return TensorValue(ts, data)
+  return _TensorValue(ts, data)
 
 
-def _read_header(f: BinaryIO) -> Header:
+def _read_header(f: BinaryIO) -> _Header:
   header = json.loads(f.readline())
   tensor_specs = [create_tensorspec(ts) for ts in header['features']]
   score_spec = create_tensorspec(header['score']) if 'score' in header else None
-  return Header(features=tensor_specs, score=score_spec)
-
-
-def pretty_print_tensor_value(tv: TensorValue):
-  print(f'{tv.spec().name}: {",".join([str(v) for v in tv])}')
+  return _Header(features=tensor_specs, score=score_spec)
 
 
 @dataclasses.dataclass(frozen=True)
-class Record:
+class ObservationRecord:
   context: str
   observation_id: int
-  feature_values: List[TensorValue]
-  score: Optional[TensorValue]
+  feature_values: List[_TensorValue]
```

Comment:
nit, up to you: ObservationRecord is returned to the user (externally visible) but types are currently _Hidden (_TensorValue).  consider moving it back to public.

Reply:
renamed it to draw even more contrast to `tf.Tensor`